packages/sitecore-jss-cli/src/create/index.ts

```diff
@@ -3,73 +3,144 @@ import fs from 'fs';
 import glob from 'glob';
 import path from 'path';
 
+interface PackageJsonConfig {
+  appName: string;
+  graphQLEndpointPath?: string;
+  sitecoreDistPath?: string;
+}
+
+interface PackageJson {
+  name: string;
+  config: PackageJsonConfig;
+}
+
 /**
- * @param {string} projectFolder
- * @param {string} name
- * @param {string} hostName
+ * Escapes a string literal for use in RegEx
+ * @param {string} literal
  */
-export function applyNameToProject(projectFolder: string, name: string, hostName: string) {
-  console.log(chalk.cyan(`Setting package name ${name}...`));
+function escapeRegex(literal: string) {
+  return literal.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
+}
 
-  const packagePath = path.join(projectFolder, 'package.json');
+/**
+ * @param {PackageJson} pkg package.json object
+ * @param {string} name App name
+ * @param {string} [replaceName] Optional token which will enable "name replacement mode". If omitted, default/conventional values are used.
+ */
+export function applyNameToPackageJson(pkg: PackageJson, name: string, replaceName?: string) {
+  pkg.name = name; // root name will never match "replaceName", so always set directly
+  pkg.config.appName = replaceName
+    ? pkg.config.appName.replace(RegExp(escapeRegex(replaceName), 'g'), name)
```

Comment:
@ambrauer Can you define a separate func?
```js
const applyNameReplacement = (string value, string replaceName) => value.replace(RegExp(escapeRegex(replaceName), 'g'), name)
```
And replace all the same entries by it

Reply:
@illiakovalenko Agree, refactored to dedicated function and added tests.